File: 2-Basic-Mathematical-Knowledge-for-Algorithms/008-Brute_Force_1/c/main.c
#include <stdio.h>

int min(int a, int b) {
    return (a < b) ? a : b;
}

int max(int a, int b) {
    return (a > b) ? a : b;
}

int main() {
    // 1. 入力処理
    int n, s;
    scanf("%d %d", &n, &s);

    // 2. 最適化された計算
    int count = 0;
    for (int r = 1; r <= n; r++) {
        int max_b = min(n, s - r);
        count += max(0, max_b);
    }

    // 3. 出力処理
    printf("%d\n", count);

    return 0;
}

File: 2-Basic-Mathematical-Knowledge-for-Algorithms/008-Brute_Force_1/rust/src/main.rs
use std::io;
use std::cmp;

fn main() {
    // 1. 入力処理
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let numbers: Vec<i32> = input
        .trim()
        .split_whitespace()
        .map(|s| s.parse().expect("Please enter a number"))
        .collect();
    
    let n = numbers[0];
    let s = numbers[1];

    // 2. 最適化された計算
    let mut count = 0;
    for r in 1..=n {
        let max_b = cmp::min(n, s - r);
        count += cmp::max(0, max_b);
    }

    // 3. 出力処理
    println!("{}", count);
}